### PR TITLE
Add functionalities to Search

### PR DESCRIPTION
**Pull request by Miyunecadz**

### CODE REVIEW

File: app/Http/Controllers/BLDGApprovalController.php [hunk 1]
```diff
@@ -7,13 +7,32 @@
...
+use Illuminate\Pagination\Paginator;
+use Illuminate\Support\Facades\DB;
 
 class BLDGApprovalController extends Controller
 {
     public function index()
     {
         $services = Service::where('status', 'pending_building_inspection')->paginate(20);
-        return view('pages.bldg-request-approval', ['route' => 'admin.undo', 'text' => ['Lists of Request for Building/Area Inspections', 'View Approved Request'], 'search_heading' => 'SEARCH REQUEST','services' => $services]);
+        return view('pages.bldg-request-approval', ['search_route' => 'admin.search','route' => 'admin.undo', 'text' => ['Lists of Request for Building/Area Inspections', 'View Approved Request'], 'search_heading' => 'SEARCH REQUEST','services' => $services]);
```
Comment (armanmasangkay) [inline app/Http/Controllers/BLDGApprovalController.php:19]: The data provided on every view is almost the same. I think we can extract a function for this that will build this array for us and just accept parameters based on the data that will change

File: app/Http/Controllers/WaterWorksApprovalController.php [hunk 2]
```diff
@@ -5,15 +5,33 @@
...
+use App\Services\ServicesFromKeyword;
+use Illuminate\Pagination\Paginator;
 
 class WaterWorksApprovalController extends Controller
 {
     public function index()
     {
         $services = Service::where('status', 'pending_waterworks_inspection')->paginate(20);
-        return view('pages.waterworks-request-approval', ['route' => 'admin.waterworks-request-approvals', 'search_heading' => 'SEARCH REQUEST', 'services' => $services]);
+        return view('pages.waterworks-request-approval', ['search_route' => 'admin.water.search','route' => 'admin.waterworks-request-approvals', 'search_heading' => 'SEARCH REQUEST', 'services' => $services]);
```
Comment (armanmasangkay) [inline app/Http/Controllers/WaterWorksApprovalController.php:16]: Same here, we can create a function that will build the data we provide to the view.

File: app/Services/ServicesFromKeyword.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?php namespace App\Services;
+
+use App\Models\Service;
+
+class ServicesFromKeyword{
+
+
+    public function get($keyword, $status)
+    {
+        $services=Service::rightjoin('customers', 'customer_id' ,'=', 'account_number')
```
Comment (armanmasangkay) [inline app/Services/ServicesFromKeyword.php:10]: Why use rightjoin?

File: resources/views/templates/editBillingModal.blade.php [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
-                <input type="hidden" name="edit_excess_rate" value="{{ isset($rates) ? $rates['excess_rates'] : '0'}}">
-                <input type="hidden" name="edit_max_range" value="{{ isset($rates) ? $rates['max_range'] : '0'}}">
-                <input type="hidden" name="edit_or_num" value="{{ isset($customer) ? ($customer['balance'] != null ? $customer['balance']->payment_or_no : '') : ''}}">
-                <input type="hidden" name="edit_surcharge" value="{{ isset($surcharge) ? $surcharge : '0'}}">
-                <input type="hidden" name="edit_customer_id" value="{{ isset($customer) ? $customer['account'] : '' }}">
-                <input type="hidden" name="edit_curr_transaction_id" value="{{ isset($current_transaction_id) ? $current_transaction_id : '' }}">
-                <input type="hidden" name="edit_prev_transaction_id" value="">
+                @if ($customer && $rates)
+                    <input type="hidden" name="edit_connection_type" value="{{ isset($customer)?$customer['connection_type'] : ''}}">
+                    <input type="hidden" name="edit_min_rates" value="{{ isset($rates) ? $rates['min_rate'] : '0'}}">
```
Comment (armanmasangkay) [inline resources/views/templates/editBillingModal.blade.php:9]: we can use `$rates['min_rate']??'0'` this is much shorter. Same goes with the other variables

**Review by armanmasangkay — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/BLDGApprovalController.php (hunk 1 above)
Comment (Miyunecadz) [reply app/Http/Controllers/BLDGApprovalController.php]: I provide a class which will return array value.

image

**Review by armanmasangkay — APPROVED** [review]

**Pull request merged**